### DISCUSSION TITLE
Encrypted cookies, rack-protection vs rack-session

### DISCUSSION THREAD

**Discussion by dentarg** [sinatra/sinatra#1878] [Q&A] [answered]
Is https://github.com/sinatra/sinatra/pull/1324 and https://github.com/rack/rack/pull/1805 (started as https://github.com/rack/rack/pull/1177) (now living in https://github.com/rack/rack-session) the same?

**Comment by jkowens** [comment]
@dentarg yep that’s what I got from those threads.

**Comment by dentarg** [comment]
Seeing they are the same, is it wise to have them duplicated going forward?

I guess what I'm asking is if the code in Sinatra should be dropped in favor of the rack-session code

**Comment by jkowens** [comment, marked as answer]
Ohh yes. Yes we should definitely drop that now that we can.
